alf/algorithms/off_policy_algorithm.py

```diff
@@ -94,9 +97,10 @@ def train(self,
         """
         if mini_batch_size is None:
             mini_batch_size = self._exp_replayer.batch_size
-        if clear_replay_buffer:
+        if whole_replay_buffer_training:
```

Comment:
because the batch_size changes. You may need to change line 122 to @common.function(experimental_relax_shapes=True). Otherwise, every call will trigger retrace, which is expensive.

Reply:
The actual size of the batch for training, which is ```mini_batch_size```` does not change:
https://github.com/HorizonRobotics/alf/blob/c59e6bfc22fc6d5c3e97aa918b110b03cdc58999/alf/algorithms/off_policy_algorithm.py#L180-L182

Do we still need to add ```@common.function(experimental_relax_shapes=True)```?
It's not very clear to be the best place to add it.

Reply:
But the shape of the input arguments to _train() changes. Do you see tensorflow display messages about retracing and/or experimental_relax_shapes?

Reply:
I didn't see info about message from tf about retracing and/or experimental_relax_shapes.